lisp/ess-r-mode.el

```diff
@@ -711,84 +711,53 @@ Executed in process buffer."
         (cons val long-path)
       val)))
 
+(defmacro ess-r-create-runner (arg)
```

Comment:
Actually after reading the rest I'm not so sure what are the semantics here. I think you have two options: a `defun`-like macro that takes a symbol (and checks for it). Or a pure function and then I'm not sure what name to give it (not `create` because it'd still define a function as a side-effect, maybe `ess-r-define-runner` without the abbrev would be less macro-like as a name).

In the `while` loop below you seem to use it more like a function than like a macro. So it's easiest to just make it a function (and then you don't have the double evaluation problem I mentioned below). If you do go for a macro you should probably wrap around `defun` instead of `fset`.

Reply:
I like `ess-defrunner`. Let's make it generic if possible.